runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -2729,15 +2729,24 @@ done:;
 		J9Class *componentClazz = J9VM_J9CLASS_FROM_HEAPCLASS(_currentThread, *(j9object_t*)_sp);
 		J9Class *arrayClazz = componentClazz->arrayClass;
 		if (NULL == arrayClazz) {
+			buildInternalNativeStackFrame(REGISTER_ARGS);
+			updateVMStruct(REGISTER_ARGS);
 			arrayClazz = internalCreateArrayClass(_currentThread, 
 				(J9ROMArrayClass *) J9ROMIMAGEHEADER_FIRSTCLASS(_currentThread->javaVM->arrayROMClasses), 
 				componentClazz);
+			VMStructHasBeenUpdated(REGISTER_ARGS);
+			if (VM_VMHelpers::exceptionPending(_currentThread)) {
```

Comment:
The `if (NULL == arrayClazz) {` should either be moved into this check as they are basically doing the same thing or turned into a Trace Assert.

Reply:
updated.